broadinstitute/ddp-study-server
# DDP-8374 | Onc History Created

## Context

Ticket: https://broadinstitute.atlassian.net/browse/DDP-8374

Problem description: When we creade the first record for `onc_history_detail` it must also set `onc_history.created` to the current date and it was already doing it correctly in our primary DB but not in ES. Also we had a bug regarding patching the existing `onc_history`

Solution: Export `onc_history.created` into ES while creating the first record for `onc_history_detail` and and change/add special solutions for patching the existing `onc_history`.

Tech details:
- Add @TableName annotation for `OncHistory.java` and also change the name in `NameValue` to `o.created` to signify that the NameValue is meant to be craeted for `OncHistory` since the alias `o` corresponds to `OncHistory`.
- Create `OncHistoryDBElementBuilder` which calls the super and then changes the primary id to be `participant_id` since that's the id we get from the client.
- Act accordingly and establish a new `ExistingOncHistoryPatch` coupled with `OncHistoryDBElementBuilder` for exercising the correct patch.
- Modify the `ExistingRecordPatchFactory.produce` static factory based on new additions.
- Write unit tests


Steps to reproduce & test yourself:

1) Go to pt list page
2) Find the pt with no `Onc History Detail`-s
3) Create first `Onc History Detail`
4) Refresh the page in order to load updated data
5) Attest that `Onc History Detail` is created along with today's date written in `Onc History Created` field 
6) Attempt to patch `Onc History Reviewed` (set some date) and attest that it works too
7) Reload the page and see the final results (new `Onc History Detail`, today's date in `Onc History Created` and some date value for `Onc History Reviewed` too.) 

_Say a few words about the bigger picture beyond this PR. How do your changes improve life for our users? Link directly to the jira ticket (just type "DDP-..." and it'll automatically link) to help your teammates quickly get up to speed on the motivation for these changes._

## Checklist

- [ ] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [ ] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

_If unsure or need help with any of the above items, add the `help wanted` label. For items that starts with `If applicable`, if it is not applicable, check it off and add `n/a` in front._

## FUD Score

_Overall, how are you feeling about these changes?_

- [ ] :sweat_smile: There might be some issues here

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [ ] They are user-visible in dev as a regular user journey and require no additional instructions.

## Testing

- [ ] I have written zero automated tests but have poked around locally to verify proper functionality

## Release
